### DISCUSSION TITLE
Data visualisation with Highcharts

### DISCUSSION THREAD

**Discussion by james2hey** [twilio-labs/paste#2841] [General] [answered]
Hello,

We are about to start creating some charts for Flex Insights and are investigating which data visualisation library to use.

We have noticed that paste supports Highcharts integration by providing the `usePasteHighchartsTheme` hook, but have also seen that there is licensing around this library to remove the watermark.

It seems like we have two options to do this integration with paste:
* Pay for the Highcharts subscription so we can integrate directly with`usePasteHighchartsTheme`, or
* Map the paste theme to a free alternative charting library (e.g. recharts, react-chart-2 etc)

Are there any additional benefits (e.g. accessibility) that Highcharts has over a free alternative that are a reason it's the chosen library to support by paste?

Or are there any plans to support other free data visualisation libraries?

Thanks!

**Comment by SiTaggart** [comment, marked as answer]
Hi @james2hey 

For charting at Twilio we have decided to go with Highcharts. The details on that decision can be found in the following ADR https://github.com/twilio-internal/architecture/blob/main/language-guilds/typescript-guild/adr/0002-charting-library.md. The ADR outlines alternatives explored and benefits of Highcharts over each.

We're still working on updated licensing for Twilio, with the idea that everyone will be able to use Highcharts on Twilio applications.

**Reply by james2hey** [reply]
Do you know if that updated licensing would be valid for customers who import our components and host them on their own application?

**Reply by SiTaggart** [reply]
I'm no legal expert, so we'd need to check, but potentially. Licensing only concerns "simultaneous" developer access to the Highcharts API. So, touching the highchart API directly.

If we start redistributing it as is, as a developer API to the Highcharts library directly, then we probably can't use it for that purpose. 

Depends on the usecase, if you are creating a chart component as an example, with an API that a developer can use to customize the chart data and colors for example, we're not technically exposing the Highcharts API. We'd just need to double check
